include/multipass/utils/permission_utils.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef MULTIPASS_PERMISSION_UTILS_H
+#define MULTIPASS_PERMISSION_UTILS_H
+
+#include <multipass/singleton.h>
+
+#include <QFileDevice>
+#include <filesystem>
+
+#define MP_PERMISSIONS multipass::PermissionUtils::instance()
+
+namespace multipass
+{
+
+class PermissionUtils : public Singleton<PermissionUtils>
+{
+public:
+    using Path = std::filesystem::path;
```

Comment:
`namespace fs = std::filesystem` is already declared elsewhere as well and has other limitations that `using` avoids. I prefer this since it's contained in the `PermissionUtils` class. Of course it'd be nice to just have one `Path` but the codebase is split between `QString` and `fs::path`.